DQMOffline/Trigger/plugins/HLTFiltersDQMonitor.cc

```diff
@@ -249,209 +286,256 @@ void HLTFiltersDQMonitor::analyze(const edm::Event& iEvent, const edm::EventSetu
+                                        << triggerEventWithRefsLabels.module << ":"
+                                        << triggerEventWithRefsLabels.productInstance << ":"
+                                        << triggerEventWithRefsLabels.process << "\")";
+
+    triggerEventWithRefs = iEvent.getHandle(triggerEventWithRefsToken_);
+    if (not triggerEventWithRefs.isValid()) {
+      edm::LogWarning("HLTFiltersDQMonitor")
+          << "invalid handle to trigger::TriggerEventWithRefs (InputTag: \"" << triggerEventWithRefsLabels.module << ":"
+          << triggerEventWithRefsLabels.productInstance << ":" << triggerEventWithRefsLabels.process
+          << "\") -> plugin will not fill DQM outputs for this event";
       return;
     }
   }
 
-  auto const triggerEventSize(useTriggerEventAOD ? triggerEventAOD->sizeFilters() : triggerEventRAW->size());
-  LogTrace("") << "[HLTFiltersDQMonitor::analyze] useTriggerEventAOD = " << useTriggerEventAOD
-               << ", triggerEventSize = " << triggerEventSize;
+  auto const triggerEventSize = useTriggerEvent ? triggerEventHandle->sizeFilters() : triggerEventWithRefs->size();
+  LogTrace("HLTFiltersDQMonitor") << "[HLTFiltersDQMonitor::analyze] useTriggerEvent = " << useTriggerEvent
+                                  << ", triggerEventSize = " << triggerEventSize;
 
   // fill MonitorElements for PrimaryDatasets and Paths
   // loop over Streams
   for (auto const& istream : hltConfigProvider_.streamNames()) {
-    LogTrace("") << "[HLTFiltersDQMonitor::analyze]   Stream = \"" << istream << "\"";
-    auto const& dsets(hltConfigProvider_.streamContent(istream));
+    LogTrace("HLTFiltersDQMonitor") << "[HLTFiltersDQMonitor::analyze]   Stream = \"" << istream << "\"";
+
     // loop over PrimaryDatasets in Stream
+    auto const& dsets = hltConfigProvider_.streamContent(istream);
     for (auto const& idset : dsets) {
-      LogTrace("") << "[HLTFiltersDQMonitor::analyze]     Dataset = \"" << idset << "\"";
-      TProfile* meDatasetTProf(nullptr);
-      const std::string meDatasetName(efficPlotNamePrefix_ + idset);
-      if (meDatasetMap_.find(meDatasetName) != meDatasetMap_.end()) {
-        meDatasetTProf = meDatasetMap_.at(meDatasetName)->getTProfile();
+      LogTrace("HLTFiltersDQMonitor") << "[HLTFiltersDQMonitor::analyze]     Dataset = \"" << idset << "\"";
+
+      // consider only Datasets with a MonitorElement (see bookHistograms)
+      auto const meDatasetName = efficPlotNamePrefix_ + idset;
+      if (meDatasetMap_.find(meDatasetName) == meDatasetMap_.end()) {
+        LogDebug("HLTFiltersDQMonitor") << "No MonitorElement associated to Dataset \"" << idset << "\" in Stream \""
+                                        << istream << "\" (will be ignored)";
+        continue;
       }
-      auto const& dsetPathNames(hltConfigProvider_.datasetContent(idset));
+      MonitorElement* const meDatasetProf = meDatasetMap_[meDatasetName];
```

Comment:
Also here storing the returned iterator from `meDatasetMap_.find(meDatasetName)` in a variable and de-referencing it would be more efficient.